jrnl/Journal.py

```diff
@@ -220,11 +223,19 @@ def filter(
         if contains:
             contains_lower = contains.casefold()
 
+        # Create datetime object for comparison below
+        # this approach allows various formats
+        if month or day or year:
+            compare_d = time.parse(f"{month or 1}/{day or 1}/{year or 1}")
```

Comment:
Careful! We have some users that use day-first formats (e.g. 31/12/2021). Could any settings like potentially invert the month and day here??

Reply:
`compare_d` is independent of the journal type and the configured dateformat—it's only used for comparison.

Reply:
Note that `time.parse` returns a `datetime` object.

Reply:
Also, what happens if a user enters something like `jrnl -month feb`?

Reply:
It should work perfectly! I accounted for it in the tests too.

Reply:
Hmm... perhaps I can change `f"{month or 1}/{day or 1}/{year or 1}"` to something clearer. In either case, it works!

Reply:
It seems like `month/day/year` is the only one `parsedatetime` parses like we need.

Reply:
Ah, okay. I'm glad you've looked at this already!

I thought this might be affected by system settings (like locale or timezone, maybe). Are you sure those don't do anything here?

Reply:
Turns out that `dateutil` is responsible for parsing this, and in debugging their code, I found a more standard format: `month.day.year`.

Regarding system settings, I doubt there is any issue: I am using jrnl's `time.parse`.

I am marking this as resolved.